filebeat/harvester/reader/message.go

```diff
@@ -0,0 +1,22 @@
+package reader
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Message represents a reader event with timestamp, content and actual number
+// of bytes read from input before decoding.
+type Message struct {
+	Ts      time.Time     // timestamp the content was read
+	Content []byte        // actual content read
+	Bytes   int           // total number of bytes read to generate the message
+	Fields  common.MapStr // optional fields that can be added by reader
+}
+
+func NewMessage() Message {
+	return Message{
+		Fields: common.MapStr{},
```

Comment:
what's the problem with `Fields` being nil by default?

Why not have a 

```
func (msg *Message) AddFields(fields common.MapStr) {
    if msg.Fields == nil {
       msg.Fields = common.Mapstr{}
    }
    msg.Fields.Update(fields)
}
```

any processor must use `AddFields` only, so some share `common.MapStr` will be updated by accident.